src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveCompletionItemProvider.cs

```diff
@@ -66,10 +66,11 @@ internal static bool AtDirectiveCompletionPoint(RazorSyntaxTree syntaxTree, Sour
                 return false;
             }
 
-            if (implicitExpression.FullWidth > 2)
+            if (implicitExpression.FullWidth > 2 && context.CompletionRequestedReason != CompletionReason.Automated)
             {
                 // We only want to provide directive completions if the implicit expression is empty "@|" or at the beginning of a word "@i|", this ensures
-                // we're consistent with how C# typically provides completion items
+                // we're consistent with how C# typically provides completion items.
+                // We also need to return results if this request was made as part of an IncompleteRequest invokation
```

Comment:
Shouldn't reference the underlying reason. We want the completion item providers to be less knowledgeable about the leaking concerns from the LSP endpoints.